### PR TITLE
cache the averaging operators

### PR DESCRIPTION
**Pull request by lheagy**
- cache `aveF2CC`, `aveF2CCV`, `aveE2CC`, `aveE2CCV` as per #48 
- question: do we want to be also caching the components (eg. `aveFx2CC`, ...)? Currently we are, but I think this might be unnecessary. The components are called once more to get the regularization operators constructed, but are then cached there, so it is really only one more call

### CODE REVIEW

File: discretize/DiffOperators.py [hunk 1]
```diff
@@ -702,11 +711,11 @@ def aveFx2CC(self):
 
         if getattr(self, '_aveFx2CC', None) is None:
             n = self.vnC
-            if(self.dim == 1):
+            if self.dim == 1:
```
Comment (rowanc1) [inline discretize/DiffOperators.py]: Thanks for cleaning things up as well. :)

**Comment by ahartikainen** [comment]
Should there be a function to reset all cached variables to None (if one edits the mesh)?

e.g.

    mesh.reset_cache()

**Comment by lheagy** [comment]
Good point @ahartikainen. The operators would only need to be reset of the cell sizes changed. Initially, I was inclined to say that if you are changing the cell-sizes, you should just be creating a new mesh, and thus we would never need to clear the operators. However, for example, with the tree mesh, you may refine, run a test or something and then choose to refine or coarsen aspects of the mesh, in which case, we should clear all cached operators. 

cc @rowanc1. 

Maybe for naming it is 
```
mesh.clean()
``` 
?

**Comment by lheagy** [comment]
I have included the discussion of `.clean()` in #48

**Pull request merged**